rustls/src/hash_hs.rs

```diff
@@ -107,12 +101,13 @@ impl HandshakeHash {
     /// Get the hash value if we were to hash `extra` too,
     /// using hash function `hash`.
     pub fn get_hash_given(&self, hash: &'static digest::Algorithm, extra: &[u8]) -> Vec<u8> {
-        let mut ctx = if self.ctx.is_none() {
-            let mut ctx = digest::Context::new(hash);
-            ctx.update(&self.buffer);
-            ctx
-        } else {
-            self.ctx.as_ref().unwrap().clone()
+        let mut ctx = match &self.ctx {
+            Some((_, ctx)) => ctx.clone(),
+            None => {
+                let mut ctx = digest::Context::new(hash);
+                ctx.update(&self.buffer);
+                ctx
```

Comment:
This is duplicating PR #451. I agree your way of using `ctx.clone()` is better assuming it is correct. However, I think that should be done in a separate PR. In this PR, I would keep the logic the same but remove the unwraps. In this specific instance, I would just revert this change in favor of PR #451.